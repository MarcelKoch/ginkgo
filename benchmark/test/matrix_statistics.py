#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.mtx.json")],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
)

# input matrix file
test_framework.compare_output(
    ["-input_matrix", str(test_framework.matrixpath)],
    expected_stdout="matrix_statistics.matrix.stdout",
    expected_stderr="matrix_statistics.matrix.stderr",
)
