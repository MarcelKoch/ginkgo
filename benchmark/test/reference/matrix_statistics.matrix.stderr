This is Ginkgo 1.7.0 (develop)
    running with core module 1.7.0 (develop)
Running test case
{
    "filename": "",
    "problem": {}
}
Matrix is of size (36, 36)
